lib/java/com/google/android/material/tabs/TabLayout.java

```diff
@@ -2898,13 +2898,16 @@ private void updateTextAndIcon(
         }
       }
 
-      if (iconView != null) {
```

Comment:
To be honest, I like my solution better because I think the condition:
> "If iconView and textView are visible (more precisely, "take up space"), there should be a gap between them" 

is simpler, clearer, and slightly more correct than

> "If iconView is visible, text is not empty, and labelVisibilityMode == TAB_LABEL_VISIBILITY_LABELED, there should be a gap between iconView and textView".

As for refactoring of L2883-L2897 (commit `15fa4db`), I completely support you :) Moreover, I would consider refactoring not only the mentioned block, but the `updateTextAndIcon` method as a whole.

Reply:
As for refactoring of L2883-L2897 (commit `15fa4db`), I completely support you :) Moreover, I would consider refactoring not only the mentioned block, but the `updateTextAndIcon` method as a whole.

Reply:
In the commit, no dependencies are created between the text view and the icon view. The modified block L2901-2925 is responsible only for the presence/absence of a gap between the views.
As for the gap, yes, its visibility depends on the visibility of the text and the icon. It's done this way because the presence of the gap is due to the designer's decision, not the data model. In other words the gap should be visible only when both the text and the icon are visible and it doesn't matter why they are visible. Generally speaking, I could have added a gap using `setShowDividers` from `TabView`, and it wouldn't be a consistency violation or something.

On the other hand, in this particular case, it's not really a matter of principle for me. If you like the solution with `showingText` better, just confirm it and I'll do so. If you are quite satisfied with the solution above - you know what to do 😉.

Reply:
If you don't mind let's change it to the way I suggested, and I'll be more confident to take the full responsibility when we send it for internal review. : )

Reply:
Ok, done.